### PR TITLE
Pods DFV compatibility for Page Builders

### LINKED ISSUE #5799: Pods PodsDFV error when extending media
**Describe the bug**
Similar or same issue as #5313.  We are seeing an issue with Beaver Builder when taking the steps below.   (Note we also saw the same when testing with Elementor)

**To Reproduce**

1. Install BB Lite
2. Pods Admin > Add New > Extend Existing
3. Set Content Type to Media
4. Just add any field and save the Pod
5. Drop a Photo module to a page then try selecting any of the images. You can click on an image but "Select Photo" is not clickable and you see the error about PodsDFV being undefined in console.

**Expected behavior**
That "Select Photo" button is clickable

**Screenshots**
Image 2020-08-07 at 12 54 32 PM

## Pods Version
2.7.21

## WordPress Environment
<details>
WordPress Version: 5.4.2

PHP Version: 7.4.6

MySQL Version: 5.5.5

Server Software: nginx/1.19.0

Your User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.105 Safari/537.36

Session Save Path:

Session Save Path Exists: No

Session Save Path Writeable: No

Session Max Lifetime: 1440

Opcode Cache:

Apc: No
Memcached: No
OPcache: Yes
Redis: No
Object Cache:

APC: No
APCu: No
Memcache: Yes
Memcached: Yes
Redis: Yes
WPDB Prefix: test_

WP Multisite Mode: No

WP Memory Limit: 40M

Current Memory Usage: 22.260M

Current Memory Usage (real): 2.000M

Pods Network-Wide Activated: No

Pods Install Location: /home/test/webapps/test/wp-content/plugins/pods/

Pods Tableless Mode Activated: No

Pods Light Mode Activated: No

Currently Active Theme: Beaver Builder Theme

Currently Active Plugins:

Beaver Builder Plugin (Lite Version): 2.3.2.5
Pods - Custom Content Types and Fields: 2.7.21

</details>

## Pods Package Export (helpful!)

```
{"meta":{"version":"2.7.21","build":1596822965},"pods":{"2776":{"id":2776,"name":"media","label":"Media","description":"","type":"media","storage":"meta","object":"media","alias":"","fields":{"text":{"id":2777,"name":"text","label":"Text","description":"","help":"","class":"","type":"text","weight":0,"pick_object":"","pick_val":"","sister_id":"","required":"0","text_allow_shortcode":"0","text_allow_html":"0","text_allowed_html_tags":"strong em a ul ol li b i","text_max_length":"255","pick_post_status":["publish"],"admin_only":"0","restrict_role":"0","restrict_capability":"0","hidden":"0","read_only":"0","roles_allowed":["administrator"],"rest_read":"0","rest_write":"0","rest_pick_response":"array","rest_pick_depth":"2","unique":"0","text_repeatable":"0","text_placeholder":""}},"show_in_menu":1,"rest_enable":"0","read_all":"0","write_all":"0"}}}
```

### PR DESCRIPTION
**Pull request by JoryHogeveen**
Fixes #5799 

TODO:
- [x] Beaver Builder

TODO Pods 2.8:
- Divi _(Successfully added Pods scripts but renders errors due to Marionette)_
- Elementor _(Successfully added Pods scripts but renders incompatible with Elementor due to Marionette)_

## PR Checklist

- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: classes/PodsInit.php [hunk 1]
```diff
@@ -554,12 +556,22 @@ public function register_assets() {
...
 				'media-models',
 			), PODS_VERSION, true
 		);
 
+		/**
+		 * Page builders.
+		 */
+		if (
+			doing_action( 'elementor/editor/before_enqueue_scripts' ) ||
+			isset( $_GET['fl_builder'] )
```
Comment (tr1b0t) [inline classes/PodsInit.php:570]: :warning: Processing form data without nonce verification.

<sub>hash: 3ce741b9ece1f502b0d693f49ecd4aba</sub>

File: classes/PodsInit.php [hunk 2]
```diff
@@ -554,12 +556,23 @@ public function register_assets() {
...
 				'media-views',
 				'media-models',
 			), PODS_VERSION, true
 		);
 
+		/**
+		 * Page builders.
+		 */
+		if (
+			doing_action( 'elementor/editor/before_enqueue_scripts' ) || // Elementor
```
Comment (tr1b0t) [inline classes/PodsInit.php:569]: :no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 705b88bf92c18c692f2ff6d885551f55</sub>

File: classes/PodsInit.php [hunk 3]
```diff
@@ -554,12 +556,23 @@ public function register_assets() {
...
+			isset( $_GET['fl_builder'] ) || // Beaver Builder
```
Comment (tr1b0t) [inline classes/PodsInit.php:570]: * :warning: Processing form data without nonce verification.
* :no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: f81017799020eef9221efa79f4480e33</sub>

File: classes/PodsInit.php [hunk 4]
```diff
@@ -554,12 +556,23 @@ public function register_assets() {
...
+			isset( $_GET['et_fb'] ) // Divi
```
Comment (tr1b0t) [inline classes/PodsInit.php:571]: * :warning: Processing form data without nonce verification.
* :no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 88a5a5e9c1fbc30ee7bab70dd505b957</sub>

**Comment by JoryHogeveen** [comment]
@sc0ttkclark It seems the noConflict fix in #5358 isn't enough to properly make our Marionette instance compatible with other plugins.

File: classes/PodsInit.php [hunk 5]
```diff
@@ -478,6 +478,8 @@ public function register_assets() {
 
 		$maybe_min = SCRIPT_DEBUG ? '' : '.min';
 
+		do_action( 'pods_before_enqueue_scripts', $maybe_min );
```
Comment (sc0ttkclark) [inline classes/PodsInit.php:481]: Needs phpdoc

File: classes/PodsInit.php [hunk 6]
```diff
@@ -598,6 +611,8 @@ public function register_assets() {
 			wp_enqueue_script( 'pods-dfv' );
 			wp_enqueue_style( 'pods-form' );
 		}
+
+		do_action( 'pods_after_enqueue_scripts', $maybe_min );
```
Comment (sc0ttkclark) [inline classes/PodsInit.php:615]: Needs phpdoc

File: classes/PodsInit.php [hunk 7]
```diff
@@ -511,7 +520,7 @@ public function register_assets() {
 
 		if ( ! wp_script_is( 'jquery-ui-timepicker', 'registered' ) ) {
 			wp_register_script(
-				'jquery-ui-timepicker', PODS_URL . 'ui/js/timepicker/jquery-ui-timepicker-addon.min.js', array(
+				'jquery-ui-timepicker', PODS_URL . "ui/js/timepicker/jquery-ui-timepicker-addon{$suffix_min}.js", array(
```
Comment (tr1b0t) [inline classes/PodsInit.php:523]: * :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: 6e7293be02d31887e0eb1615686c493e</sub>

File: classes/PodsInit.php [hunk 8]
```diff
@@ -521,29 +530,29 @@ public function register_assets() {
 			);
 		}
 		if ( ! wp_style_is( 'jquery-ui-timepicker', 'registered' ) ) {
-			wp_register_style( 'jquery-ui-timepicker', PODS_URL . 'ui/js/timepicker/jquery-ui-timepicker-addon.min.css', array(), '1.6.3' );
+			wp_register_style( 'jquery-ui-timepicker', PODS_URL . "ui/js/timepicker/jquery-ui-timepicker-addon{$suffix_min}.css", array(), '1.6.3' );
 		}
 
 		wp_register_script(
-			'pods-select2', PODS_URL . "ui/js/selectWoo/selectWoo{$maybe_min}.js", array(
+			'pods-select2', PODS_URL . "ui/js/selectWoo/selectWoo{$suffix_min}.js", array(
```
Comment (tr1b0t) [inline classes/PodsInit.php:537]: * :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: 9d971c0eea2b3ef8a71c2454e73e11b6</sub>

File: classes/PodsInit.php [hunk 9]
```diff
@@ -485,13 +494,16 @@ public function register_assets() {
 		}
 
 		wp_register_script(
-			'pods', PODS_URL . 'ui/js/jquery.pods.js', array(
+			'pods', PODS_URL . 'ui/js/jquery.pods.js',
```
Comment (tr1b0t) [inline classes/PodsInit.php:497]: :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: bf47de6c732ec4001373c6a3879cecff</sub>

File: classes/PodsInit.php [hunk 10]
```diff
@@ -504,69 +516,102 @@ public function register_assets() {
 		wp_register_script( 'pods-codemirror-mode-html', PODS_URL . 'ui/js/codemirror/mode/htmlmixed/htmlmixed.js', array( 'pods-codemirror' ), '4.8', true );
 		wp_register_script( 'pods-codemirror-mode-css', PODS_URL . 'ui/js/codemirror/mode/css/css.js', array( 'pods-codemirror' ), '4.8', true );
 
+		// jQuery Timepicker.
 		if ( ! wp_script_is( 'jquery-ui-slideraccess', 'registered' ) ) {
 			// No need to add dependencies. All managed by jquery-ui-timepicker.
-			wp_register_script( 'jquery-ui-slideraccess', PODS_URL . 'ui/js/timepicker/jquery-ui-sliderAccess.js', array(), '0.3' );
+			wp_register_script('jquery-ui-slideraccess', PODS_URL . 'ui/js/timepicker/jquery-ui-sliderAccess.js', array(), '0.3' );
```
Comment (tr1b0t) [inline classes/PodsInit.php:522]: * :warning: In footer ($in_footer) is not set explicitly wp_register_script; It is recommended to load scripts in the footer. Please set this value to `true` to load it in the footer, or explicitly `false` if it should be loaded in the header.
* :no_entry_sign: Expected 1 spaces after opening parenthesis; 0 found
* :no_entry_sign: There must be a single space after the first parenthesis

<sub>hash: 388fc4b2b61c3db83268e889cc25a2b9</sub>

File: classes/PodsInit.php [hunk 11]
```diff
@@ -504,69 +516,102 @@ public function register_assets() {
...
 		}
-
 		if ( ! wp_script_is( 'jquery-ui-timepicker', 'registered' ) ) {
 			wp_register_script(
-				'jquery-ui-timepicker', PODS_URL . 'ui/js/timepicker/jquery-ui-timepicker-addon.min.js', array(
+				'jquery-ui-timepicker', PODS_URL . "ui/js/timepicker/jquery-ui-timepicker-addon{$suffix_min}.js",
```
Comment (tr1b0t) [inline classes/PodsInit.php:526]: :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: 70454ffa6725a9a9674e005e94ba2267</sub>

File: classes/PodsInit.php (hunk 10 above)
Comment (sc0ttkclark) [inline classes/PodsInit.php:522]:
```suggestion
			wp_register_script( 'jquery-ui-slideraccess', PODS_URL . 'ui/js/timepicker/jquery-ui-sliderAccess.js', array(), '0.3' );
```

File: classes/PodsInit.php (hunk 11 above)
Comment (sc0ttkclark) [reply classes/PodsInit.php:526]:
```suggestion
				'jquery-ui-timepicker',
				PODS_URL . "ui/js/timepicker/jquery-ui-timepicker-addon{$suffix_min}.js",
```

File: classes/PodsInit.php (hunk 9 above)
Comment (sc0ttkclark) [reply classes/PodsInit.php:497]:
```suggestion
			'pods',
			PODS_URL . 'ui/js/jquery.pods.js',
```

**Review by sc0ttkclark — APPROVED** [review]

File: classes/PodsI18n.php [hunk 12]
```diff
@@ -36,9 +36,7 @@ private function __construct() {
 		self::$instance = $this;
 
 		// Hook all enqueue scripts actions
-		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
-		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
-		add_action( 'login_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+		add_action( 'pods_before_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
```
Comment (tr1b0t) [inline classes/PodsI18n.php:39]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 74c4e3fb6be84e8c265dfa133b7ee9cf</sub>

File: classes/PodsInit.php [hunk 13]
```diff
@@ -485,13 +494,17 @@ public function register_assets() {
 		}
 
 		wp_register_script(
-			'pods', PODS_URL . 'ui/js/jquery.pods.js', array(
+			'pods',
+			PODS_URL . 'ui/js/jquery.pods.js',
+			array(
```
Comment (tr1b0t) [inline classes/PodsInit.php:499]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 625bc5885b16b1d99b5641aa04231868</sub>

File: classes/PodsInit.php [hunk 14]
```diff
@@ -504,69 +517,103 @@ public function register_assets() {
 		wp_register_script( 'pods-codemirror-mode-html', PODS_URL . 'ui/js/codemirror/mode/htmlmixed/htmlmixed.js', array( 'pods-codemirror' ), '4.8', true );
 		wp_register_script( 'pods-codemirror-mode-css', PODS_URL . 'ui/js/codemirror/mode/css/css.js', array( 'pods-codemirror' ), '4.8', true );
 
+		// jQuery Timepicker.
 		if ( ! wp_script_is( 'jquery-ui-slideraccess', 'registered' ) ) {
 			// No need to add dependencies. All managed by jquery-ui-timepicker.
 			wp_register_script( 'jquery-ui-slideraccess', PODS_URL . 'ui/js/timepicker/jquery-ui-sliderAccess.js', array(), '0.3' );
```
Comment (tr1b0t) [inline classes/PodsInit.php:523]: * :warning: In footer ($in_footer) is not set explicitly wp_register_script; It is recommended to load scripts in the footer. Please set this value to `true` to load it in the footer, or explicitly `false` if it should be loaded in the header.
* :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 948faccadda26110f6f0efd54798382d</sub>

File: classes/PodsInit.php [hunk 15]
```diff
@@ -504,69 +517,103 @@ public function register_assets() {
...
 		}
-
 		if ( ! wp_script_is( 'jquery-ui-timepicker', 'registered' ) ) {
 			wp_register_script(
-				'jquery-ui-timepicker', PODS_URL . 'ui/js/timepicker/jquery-ui-timepicker-addon.min.js', array(
+				'jquery-ui-timepicker',
+				PODS_URL . "ui/js/timepicker/jquery-ui-timepicker-addon{$suffix_min}.js",
+				array(
```
Comment (tr1b0t) [inline classes/PodsInit.php:529]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 2aeb38690f9e917960c25348ed7d5208</sub>

File: classes/PodsInit.php [hunk 16]
```diff
@@ -504,69 +517,103 @@ public function register_assets() {
...
+				),
+				'1.6.3'
 			);
 		}
 		if ( ! wp_style_is( 'jquery-ui-timepicker', 'registered' ) ) {
-			wp_register_style( 'jquery-ui-timepicker', PODS_URL . 'ui/js/timepicker/jquery-ui-timepicker-addon.min.css', array(), '1.6.3' );
+			wp_register_style(
+				'jquery-ui-timepicker',
+				PODS_URL . "ui/js/timepicker/jquery-ui-timepicker-addon{$suffix_min}.css",
+				array(),
```
Comment (tr1b0t) [inline classes/PodsInit.php:543]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: bbc4a283473f9272271f06336c1ab1a2</sub>

File: classes/PodsInit.php [hunk 17]
```diff
@@ -504,69 +517,103 @@ public function register_assets() {
...
+				'1.6.3'
+			);
 		}
 
+		// Select 2.
 		wp_register_script(
-			'pods-select2', PODS_URL . "ui/js/selectWoo/selectWoo{$maybe_min}.js", array(
+			'pods-select2',
+			PODS_URL . "ui/js/selectWoo/selectWoo{$suffix_min}.js",
+			array(
```
Comment (tr1b0t) [inline classes/PodsInit.php:552]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 9925ef80e681abdd26b484a096f4a7b5</sub>

File: classes/PodsInit.php [hunk 18]
```diff
@@ -504,69 +517,103 @@ public function register_assets() {
...
 				'jquery',
 				'pods-i18n',
-			), '1.0.1'
+			),
+			'1.0.1'
+		);
+		wp_register_style(
+			'pods-select2',
+			PODS_URL . "ui/js/selectWoo/selectWoo{$suffix_min}.css",
+			array(),
```
Comment (tr1b0t) [inline classes/PodsInit.php:561]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 7124d1e8a2468d86bdfb04c972d2620e</sub>

File: classes/PodsInit.php [hunk 19]
```diff
@@ -504,69 +517,103 @@ public function register_assets() {
...
 		);
-		wp_register_style( 'pods-select2', PODS_URL . "ui/js/selectWoo/selectWoo{$maybe_min}.css", array(), '1.0.2' );
 
-		// Marionette dependencies for MV fields
-		wp_register_script( 'backbone.radio', PODS_URL . 'ui/js/marionette/backbone.radio.min.js', array( 'backbone' ), '2.0.0', true );
+		// Marionette dependencies for DFV/MV fields.
+		wp_register_script(
+			'backbone.radio',
+			PODS_URL . "ui/js/marionette/backbone.radio{$suffix_min}.js",
+			array( 'backbone' ),
```
Comment (tr1b0t) [inline classes/PodsInit.php:569]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 2bc3164d2fdc8e1e67861ad6d331738e</sub>

File: classes/PodsInit.php [hunk 20]
```diff
@@ -1875,6 +1931,8 @@ public function run() {
 		add_action( 'wp_enqueue_scripts', array( $this, 'register_assets' ), 15 );
 		add_action( 'admin_enqueue_scripts', array( $this, 'register_assets' ), 15 );
 		add_action( 'login_enqueue_scripts', array( $this, 'register_assets' ), 15 );
+		// Elementor Page Builder.
+		add_action( 'elementor/editor/before_enqueue_scripts', array( $this, 'register_assets' ), 15 );
```
Comment (tr1b0t) [inline classes/PodsInit.php:1935]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: c84c972a976cba46aa6bdc38a86b3b85</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 9 error(s)
:warning: 1 warning(s)

**Review by sc0ttkclark — APPROVED** [review]

**Comment by JoryHogeveen** [comment]
@sc0ttkclark 
If you want to merge this we should omit the current Elementor part. Elementor is simply not compatible yet. Maybe it will be possible for 2.8 when we drop the DFV fields.

**Comment by sc0ttkclark** [comment]
@JoryHogeveen Sounds good on leaving out Elementor from this since it won't pass QA.

**Comment by JoryHogeveen** [comment]
@sc0ttkclark 
Divi compatibility also creates JS errors due to backbone.
It still seems to work but maybe it's better to keep this PR Beaver Builder only and fix the other builders in Pods 2.8?

**Review by sc0ttkclark — DISMISSED** [review]

**Review by sc0ttkclark — APPROVED** [review]

**Pull request merged**